src/set_timeout.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+#
+# Copyright (C) 2022 Collabora Limited
+# Author: Jeny Sadadia <jeny.sadadia@collabora.com>
+
+import logging
+import os
+import sys
+from datetime import datetime, timedelta
+from time import sleep
+
+import kernelci
+import kernelci.config
+import kernelci.db
+from kernelci.cli import Args, Command, parse_opts
+
+from logger import Logger
+
+
+class SetTimeout:
+
+    def __init__(self, configs, args):
+        self._db_config = configs['db_configs'][args.db_config]
+        api_token = os.getenv('API_TOKEN')
+        self._db = kernelci.db.get_db(self._db_config, api_token)
+        self._logger = Logger("config/logger.conf", "set_timeout")
+        self._poll_period = args.poll_period
+
+    def _update_pending_child(self, parent_id):
+        """Set child node status to timeout when parent is timed out"""
+        child_nodes = self._db.get_nodes({"status": "pending",
+                                          "parent": parent_id})
+        for child in child_nodes:
+            child['status'] = "timeout"
+            self._db.submit({'node': child})
```

Comment:
This should be recursive, child nodes can have child nodes too.

Reply:
Yes, missed it.

Reply:
Note: It's the kind of thing we could then implement on the API side for efficiency, but for now let's just keep the logic entirely on the client side as it's easier to experiment this way.